gensim/similarities/termsim.py

```diff
@@ -156,32 +140,288 @@ def most_similar(self, t1, topn=10):
                     yield (t2, similarity**self.exponent)
 
 
+def _shortest_uint_dtype(max_value):
+    """Get the shortest unsingned integer data-type required for representing values up to a given
+    maximum value.
+
+    Returns the shortest unsingned integer data-type required for representing values up to a given
+    maximum value.
+
+    Parameters
+    ----------
+    max_value : int
+        The maximum value we wish to represent.
+
+    Returns
+    -------
+    data-type
+        The shortest unsigned integer data-type required for representing values up to a given
+        maximum value.
+    """
+    if max_value < 2**8:
+        return np.uint8
+    elif max_value < 2**16:
+        return np.uint16
+    elif max_value < 2**32:
+        return np.uint32
+    return np.uint64
+
+
+def _create_source(index, dictionary, tfidf, symmetric, dominant, nonzero_limit, dtype):
+    """Build a sparse term similarity matrix using a term similarity index.
+
+    Returns
+    -------
+    matrix : :class:`scipy.sparse.coo_matrix`
+        The sparse term similarity matrix.
+
+    """
+    assert isinstance(index, TermSimilarityIndex)
+    assert dictionary is not None
+    matrix_order = len(dictionary)
+
+    if matrix_order == 0:
+        raise ValueError('The provided dictionary is empty')
+
+    logger.info("constructing a sparse term similarity matrix using %s", index)
+
+    if nonzero_limit is None:
+        nonzero_limit = matrix_order
+
+    def tfidf_sort_key(term_index):
+        if isinstance(term_index, tuple):
+            term_index, *_ = term_index
```

Comment:
I'm not familiar with this `*_` syntax – what does it do?

Reply:
It's [PEP 3132](https://www.python.org/dev/peps/pep-3132) (Extended Iterable Unpacking), here synonymous with `term_index = term_index[0]`.  
What's curious is that although this syntax is not available until after Python 3.0, the CI is passing.